flytekit/tools/fast_registration.py

```diff
@@ -139,9 +141,21 @@ def fast_package(
                         filter=lambda x: tar_strip_file_attributes(x),
                     )
 
-            with gzip.GzipFile(filename=archive_fname, mode="wb", mtime=0) as gzipped:
-                with open(tar_path, "rb") as tar_file:
-                    gzipped.write(tar_file.read())
+            if shutil.which("pigz"):
+                subprocess.run([f"pigz -c {tar_path} > {archive_fname}"], shell=True, check=True)
+            else:
+                start_time = time.time()
+                with gzip.GzipFile(filename=archive_fname, mode="wb", mtime=0) as gzipped:
+                    with open(tar_path, "rb") as tar_file:
+                        gzipped.write(tar_file.read())
+
+                end_time = time.time()
+                warning_time = 30
```

Comment:
```suggestion
                warning_time = 10
```

Reply:
Already set to 10s in this commit: https://github.com/flyteorg/flytekit/pull/2729/commits/908e1a6161bc61aec76d40386ca410c82f785b0c

Reply:
Hmm, you change that at line 192. Don't we need to change it here as well?

Reply:
You are right, sorry. Now obsolete as the logic exists only a single time in a helper function https://github.com/flyteorg/flytekit/pull/2729/commits/d32822314c401205ee93b7801dcbd53af9301775